### ISSUE TITLE
[MAINT]: Update version of go-github

### ISSUE THREAD

**Issue by grahamhar** [integrations/terraform-provider-github#2358]
### Describe the need

New features have been added to go-github and released in V65, I will raise a PR to implement this.

### SDK Version

go-github v65

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by zepeng811** [comment]
@grahamhar v65 is just released and I want to upgrade it to unblock a feature we've been waiting for a long time. 

I don't want to step over the work you've already done, do you mind updating your existing PR to v65 instead? https://github.com/integrations/terraform-provider-github/pull/2359

I confirmed that there should be no breaking changes from v64 -> v65 for this repository

**Comment by grahamhar** [comment]
> @grahamhar [v65 was just released](https://github.com/google/go-github/releases/tag/v65.0.0) and I want to upgrade it to unblock a feature we've been waiting for a long time.
> 
> I don't want to step over the work you've already done, do you mind updating your existing PR to v65 instead? #2359
> 
> I confirmed that there should be no breaking changes from v64 -> v65 for this repository

I have just quickly done that, not had a chance to run tests but a build ran cleanly.
